Article header style tweak

cheywood opened the pull request:
I've had a go at making a minor style tweak to the top area of the article. There's no issue created for these changes -  sorry if this is presumptuous or you'd rather a different workflow! Let me know.

These changes:
- Use material icons for stats at the top of the article, ala Wallabag web. It no longer displays the pixelated globe icon for the domain and the icon/s now better match the theme (no white square background for dark themes).
- Slightly increase spacing in header area. Things seemed a bit cramped to me (in these days on these bigger devices with all this space). This also provides a clearer separation between header/stats and article content. Again, now closer to the web design.

There's a mechanism in use to retain the reading of the estimated time label for TTS, and an attempt has been made to transfer the existing translations for that. A shared CSS file has been added, but I can see that doing it this way reduces the capacity to theme those items (but reduces lots of duplication). Let me know and I can duplicate more of it if that's preferred. The icon font declaration being shared I think at least makes sense.

After
![image](https://user-images.githubusercontent.com/31988069/132267890-e4a2fd2f-402a-4cbd-a9a0-a60a8d8cabf8.png)

Before
![Screenshot_20210907-095341_wallabag](https://user-images.githubusercontent.com/31988069/132267912-b03280d9-61af-4298-aa91-8bf188dc4e39.png)

After
![Screenshot_20210907-095533_wallabag](https://user-images.githubusercontent.com/31988069/132267935-04685e37-91b2-4b45-8637-1f5f88015619.png)

Before
![Screenshot_20210907-095530_wallabag](https://user-images.githubusercontent.com/31988069/132267944-3a23478c-7012-4d2f-bb76-c4da9c737093.png)

di72nn commented:
Thank you, that's a welcome contribution!
I'll try to have a look at the implementation soon.

cheywood commented:
Great, thanks!

di72nn commented on app/src/main/res/raw/webview_htmlbase.html:
```diff
@@ -13,15 +14,23 @@
 				<div id="article">
 					<header class="mbm">
 						<h1>%4$s</h1>
-						<img id="domainimg" alt="%5$s" src="file:///android_asset/ic_action_web_site.png" />
```
`ic_action_web_site.png` can be removed.

di72nn commented on app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java:
```diff
@@ -938,36 +939,47 @@ private String getExtraHead() {
+            stats.append("<li>");
+            stats.append("\t<i class=\"material-icons no-tss\">today</i>");
+            stats.append(android.text.format.DateFormat.getDateFormat(this).format(publishedAt))
+                 .append(' ')
+                 .append(android.text.format.DateFormat.getTimeFormat(this).format(publishedAt));
```
This is a very minor issue, but the indentation here is wrong (5 spaces (compared to the previous line) instead of 8).

di72nn commented on app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java:
```diff
@@ -938,36 +939,47 @@ private String getExtraHead() {
+        stats.append("<li>");
+        stats.append("\t<i class=\"material-icons no-tss\">timer</i>");
+        stats.append("<span class=\"tts-only\">")
+             .append(getString(R.string.content_estimatedReadingTimeTTSLabel))
+             .append("</span>");
```
Indentation.

di72nn commented on app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java:
```diff
@@ -976,27 +988,33 @@ private String getHeader() {
-                header.append("</div>");
+                tags.append("<div class=\"tag\">");
+                tags.append("<a href=\"tag://").append(tag.getId()).append("\">")
+                     .append(escapeHtml(tag.getLabel()))
+                     .append("</a>");
```
Indentation.

di72nn commented on app/src/main/assets/shared.css:
```diff
@@ -0,0 +1,65 @@
+  font-family: 'Material Icons';
+  font-style: normal;
+  font-weight: 400;
+  src: url(file:///android_res/raw/materialicons_regular.ttf) format('truetype');
```
I would prefer to move the file to `assets` and simply use `url('materialicons_regular.ttf')` here.

di72nn reviewed (changes requested):
Sorry for the delay.

Apart from other remarks, this seemed not to work on older Android versions: I tried it on Android 5, 6, and 7 emulators. Didn't find anything useful in logs.
Having the font in `assets` may make a difference, but I only tested it on some of these Android versions (didn't work either).

cheywood commented:
> Sorry for the delay.
> 
> Apart from other remarks, this seemed not to work on older Android versions: I tried it on Android 5, 6, and 7 emulators. Didn't find anything useful in logs. Having the font in `assets` may make a difference, but I only tested it on some of these Android versions (didn't work either).

Sorry for the slow response. Just letting you know that I will get onto this, I just got buried in other things. Hopefully time soon.

Strubbl commented on app/src/main/res/raw/tts_parser.js:
```diff
@@ -40,7 +40,9 @@ function traverse(element, callback) {
-    return element.tagName === 'SCRIPT' || element.tagName === 'NOSCRIPT';
+    return element.tagName === 'SCRIPT' || element.tagName === 'NOSCRIPT'
+        // Skip stats icons
+        || (element.tagName === 'I' && element.classList.contains('no-tss'));
```
no-tss is a typo and should be named "no-tts", or?

Strubbl commented on app/src/main/res/raw/webview_htmlbase.html:
```diff
@@ -13,15 +14,23 @@
+                        <div class="tools">
+                          <ul class="stats">
+                            <li>
+                              <i class="material-icons no-tss">link</i>
```
no-tss -> no-tts

Strubbl reviewed (changes requested):
this is very nice improvement of the article view. I found a typo only

Strubbl commented:
To have a more complete picture of the article view, can you also please check that article's tags are displayed correctly? in your screenshots the articles have no tags assigned. Maybe you can add some for testing?

cheywood commented:
Thanks @Strubbl, good pickup!

I've just pushed changes aimed at addressing all the raised issues from both of you. Older device fix is described in the commit message. Also when creating the screenshots with tags below I made some minor tweaks to the tag styling, adding a little more spacing, ala the rest of the PR.

After
![Screenshot_20211102-115639_wallabag](https://user-images.githubusercontent.com/31988069/139777413-af81c38b-7ebb-43ce-84c2-c4f85d7eea69.png)

Before
![Screenshot_20211102-115636_wallabag](https://user-images.githubusercontent.com/31988069/139777429-7685267a-3195-40c6-880d-e087f47a1ead.png)

Strubbl commented:
This PR just needs to be rebased. Do you need help with that?

cheywood commented:
No, thanks @Strubbl that should be rebased now. Let me know if you want it squashed (or I'm fine if that'd done while merging).

di72nn reviewed (changes requested):
Thanks for addressing the issues! And sorry for the delay.

It seems something broke the TTS parsing:
```
D/WebView.onCM: Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'. @ 257: file:///android_asset/
D/WebViewText: onDocumentParseStart()
V/WebViewText: onDocumentParseText(<Article title text>, 57.000000, 296.000092)
```
Can you please look into it?

`app/src/main/res/raw/materialicons_regular.ttf` should be removed now.

And what is the source of `app/src/main/assets/MaterialIcons-Regular.ttf`? The previous file is [from here](https://github.com/google/material-design-icons/blob/ab12f16d050ecb1886b606f08825d24b30acafea/font/MaterialIcons-Regular.ttf) (it got updated recently, BTW), but the new one is half the size compared to it - does it need any special treatment?

I would squash most of the stuff into the first commit, leaving only the "Attempt to update translations ..." and "Minor tweaks to tags styling in article" commits separate.

cheywood commented:
Hi, sorry again for the delay this end.

The TTS parsing issue is now fixed, `app/src/main/res/raw/materialicons_regular.ttf` removed and commits squashed.

No special treatment for the font file but it looks like I was somehow using the version from the Material Icons v3.0.1 release (maybe [from here](http://google.github.io/material-design-icons/#getting-icons), that seems to point to the older release). I've now updated it to the version in the latest release (v4.0.0).

Strubbl commented:
LGTM, thank you :rocket:

di72nn reviewed (approved):
Looks good.

// I updated the second commit to remove two `articleContent_globeIconAltText` strings from new translations.

di72nn commented:
Thanks a lot!

cheywood commented:
My pleasure! Sorry again for all the delays and relative looseness :slightly_smiling_face:
